MSAL/src/requests/MSALWebAuthResponse.m

```diff
@@ -1,139 +0,0 @@
-
-#import "MSALWebAuthResponse.h"
-#import "MSALHttpResponse.h"
-#import "MSALError.h"
-
-@interface MSALWebAuthResponse()
-
-{
-    MSALWebAuthRequest *_request;
-    id<MSALRequestContext> _context;
-}
-
-@end
-
-
-@implementation MSALWebAuthResponse
-
-NSString *const s_kWwwAuthenticateHeader = @"Accept";
-
-+ (void)processResponse:(MSALHttpResponse *)response
-                request:(MSALWebAuthRequest *)request
-                context:(id<MSALRequestContext>)context
-      completionHandler:(MSALHttpRequestCallback)completionHandler
-{
-    MSALWebAuthResponse *webAuthResponse = [MSALWebAuthResponse new];
-    webAuthResponse->_request = request;
-    webAuthResponse->_context = context;
-    
-    [webAuthResponse handleResponse:response
-                  completionHandler:completionHandler];
-    
-}
-
-
-- (id)init
-{
-    if (!(self = [super init]))
-    {
-        return nil;
-    }
-    
-    return self;
-}
-
-
-- (void)handleResponse:(MSALHttpResponse *)response
-     completionHandler:(MSALHttpRequestCallback)completionHandler
-{
-    switch (response.statusCode) {
-        case 200:
-            completionHandler(response, nil);
-            break;
-        
-        case 400:
-        case 401:
-        {
-#if PKEYAUTH_IMPLEMENTED
-            NSString *wwwAuthValue = [response.headers valueForKey:s_kWwwAuthenticateHeader];
-            
-            @throw @"to-do";
-#endif
-            completionHandler(response, nil);
-            break;
-        }
-            
-        case 500:
-        case 503:
-        case 504:
-        {
-            // retry if it is a server error
-            // 500, 503 and 504 are the ones we retry
-            if (_request.retryIfServerError)
-            {
-                _request.retryIfServerError = NO;
-
-                // retry once after hald second
-                dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.5 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
-                    [_request resend:completionHandler];
-                });
-                return;
-            }
-            //no "break;" here
-            //will go to default for handling if "retryIfServerError" is NO
-        }
-        default:
-        {
-            NSString *body = [[NSString alloc] initWithData:response.body encoding:NSUTF8StringEncoding];
-            NSString *errorData = [NSString stringWithFormat:@"Full response: %@", body];
-            
-            NSString* message = [NSString stringWithFormat:@"Error raised: (Domain: \"%@\" Response Code: %ld \n", @"Domain", (long)response.statusCode];
-            
-            NSString* messagePII = [NSString stringWithFormat:@"Error raised: (Domain: \"%@\" Response Code: %ld \n%@", @"Domain", (long)response.statusCode, errorData];
-            
-            NSMutableDictionary *userInfo = [@{MSALHTTPResponseCodeKey : [NSString stringWithFormat: @"%ld", (long)response.statusCode]} mutableCopy];
```

Comment:
Just double checking: does the new network stack have the feature to return http response code and headers?

Reply:
Currently nope. But I have added them, they will be in another PR :)